pelican/urlwrappers.py

```diff
@@ -52,27 +54,30 @@ def as_dict(self):
     def __hash__(self):
         return hash(self.slug)
 
-    def _key(self):
-        return self.slug
-
     def _normalize_key(self, key):
         subs = self.settings.get('SLUG_SUBSTITUTIONS', ())
         return six.text_type(slugify(key, subs))
 
     def __eq__(self, other):
-        return self._key() == self._normalize_key(other)
+        if isinstance(other, URLWrapper):
```

Comment:
while the point you are trying to make is valid, that differs from how pelican is currently behaving:
```

```
more: it would still evaluate to true as the normalization will return true later, although slower